alpha-build-machinery/make/operator.example.mk

```diff
@@ -22,12 +22,15 @@ CODEGEN_GROUPS_VERSION :=openshiftapiserver:v1alpha1
 #   $ make -n --print-data-base | grep ^CODEGEN
 
 # This will call a macro called "build-image" which will generate image specific targets based on the parameters:
-# $0 - macro name
-# $1 - target suffix
-# $2 - Dockerfile path
-# $3 - context directory for image build
+# $1 - target name
+# $2 - image registry
+# $3 - image org
+# $4 - image name
+# $5 - image tag
+# $6 - Dockerfile path
+# $7 - context
 # It will generate target "image-$(1)" for builing the image an binding it as a prerequisite to target "images".
-$(call build-image,origin-cluster-openshift-apiserver-operator,./Dockerfile,.)
+$(call build-image,ocp-openshift-apiserver-operator,registry.svc.ci.openshift.org,ocp,4.2,openshift-apiserver-operator,./Dockerfile.rhel,.)
```

Comment:
this is a bad interface with 7 params. Can't we combine them at the calling site? `$(addsuffix /,$(2))$(addsuffix /,$(3))$(4)$(addprefix :,$(5))` for example

Reply:
I guess, it gave us more flexibility in naming targets automatically, but now that we target as arg we can collapse it